CMakeLists.txt

```diff
@@ -24,13 +25,26 @@ cmake_dependent_option(LIBUV_BUILD_BENCH
   "Build the benchmarks when building unit tests and we are the root project" ON
   "LIBUV_BUILD_TESTS" OFF)
 
-if(MSVC)
-  list(APPEND uv_cflags /W4)
-elseif(CMAKE_C_COMPILER_ID MATCHES "AppleClang|Clang|GNU")
-  list(APPEND uv_cflags -Wall -Wextra -Wstrict-prototypes)
-  list(APPEND uv_cflags -Wno-unused-parameter)
+# Unfortunately, MSVC and clang-cl both support -Wall, but we don't actually
+# want that, as its equivalent to -Weverything :/
+if (MSVC OR CMAKE_C_SIMULATED_ID STREQUAL "MSVC")
+  check_c_compiler_flag(/W4 UV_LINT_DEFAULT)
+else()
+  check_c_compiler_flag(-Wall UV_LINT_DEFAULT)
 endif()
 
+# TODO: Place these into its own function
```

Comment:
Because...?

Reply:
It's regarding an earlier conversion with @saghul further up. Basically there's a lot of boilerplate just to "check and add" a compiler flag, and this could be reduced down to a single function.

Reply:
(I intend to have that function written before it gets merged, if that helps 😅)